pkg/discovery/task/task.go

```diff
@@ -73,22 +87,30 @@ func (t *Task) WithCluster(cluster *repository.ClusterSummary) *Task {
 
 // Get node list from the task.
 func (t *Task) NodeList() []*api.Node {
-	return t.nodeList
+	return t.nodes
 }
 
 // Get pod list from the task.
 func (t *Task) PodList() []*api.Pod {
-	return t.podList
+	return t.pods
+}
```

Comment:
As I understand, `pods` here are only set when calling `WithPods` func. And `task.WithPods` func was only called in `Dispatch` func in `dispatcher.go`. So looks this `PodList()` func will always return empty pod list in the places where `PodList()` func is called, for example in `metrics_collector.go`, `group_metrics_collector.go`, `controller_metrics_collector.go`. Maybe we can just remove this func and use `RunningPodList()` in those metrics_collectors?